tfe/resource_tfe_organization_token_test.go

```diff
@@ -99,6 +99,67 @@ func TestAccTFEOrganizationToken_existsWithForce(t *testing.T) {
 	})
 }
 
+func TestAccTFEOrganizationToken_existsWithoutExpiry(t *testing.T) {
+	token := &tfe.OrganizationToken{}
+	rInt := rand.New(rand.NewSource(time.Now().UnixNano())).Int()
+	orgName := fmt.Sprintf("tst-terraform-%d", rInt)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckTFEOrganizationTokenDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccTFEOrganizationToken_basic(rInt),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckTFEOrganizationTokenExists(
+						"tfe_organization_token.foobar", token),
+					resource.TestCheckResourceAttr(
+						"tfe_organization_token.foobar", "organization", orgName),
+				),
+			},
+
+			{
+				Config:      testAccTFEOrganizationToken_existsWithoutExpiry(rInt),
+				ExpectError: regexp.MustCompile(`must be a valid date or time`),
+			},
+		},
+	})
+}
+
+func TestAccTFEOrganizationToken_existsWithExpiry(t *testing.T) {
+	token := &tfe.OrganizationToken{}
+	rInt := rand.New(rand.NewSource(time.Now().UnixNano())).Int()
+	orgName := fmt.Sprintf("tst-terraform-%d", rInt)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckTFEOrganizationTokenDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccTFEOrganizationToken_basic(rInt),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckTFEOrganizationTokenExists(
+						"tfe_organization_token.foobar", token),
+					resource.TestCheckResourceAttr(
+						"tfe_organization_token.foobar", "organization", orgName),
+				),
+			},
+
+			{
+				Config: testAccTFEOrganizationToken_existsWithExpiry(rInt),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckTFEOrganizationTokenExists(
+						"tfe_organization_token.expiry", token),
+					resource.TestCheckResourceAttr(
+						"tfe_organization_token.expiry", "organization", orgName),
```

Comment:
I think it would make sense here to check the the expiry value from the configuration has been set on the created `resource`